subsys/net/lib/openthread/platform/radio_spinel.cpp

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright (c) 2022, Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread platform abstraction
+ *   for radio communication using Spinel.
+ *
+ */
+#include <openthread-system.h>
+#include <openthread/platform/radio.h>
+#include <zephyr/fatal.h>
+#include <zephyr/sys/util_macro.h>
+
+#include <lib/spinel/radio_spinel.hpp>
+
+#ifdef CONFIG_OPENTHREAD_HOSTPROCESSOR_SPI
+#include "spi_interface.hpp"
+
+typedef ot::Spinel::RadioSpinel<SpiInterface, RadioSpinelContext> PlatformSpiRadioSpinel;
+#else
+#error "Unsupported OpenThread Host transport type"
+#endif
+
+OT_DEFINE_ALIGNED_VAR(gRadioSpinalInstanceRaw, sizeof(PlatformSpiRadioSpinel), uint64_t);
+
+static PlatformSpiRadioSpinel& RadioSpinelInstance(void)
+{
+	void *instance = &gRadioSpinalInstanceRaw;
+
+	return *static_cast <PlatformSpiRadioSpinel *> (instance);
+}
+
+static void InitPlatformRadioSpinelInstance(void)
+{
+    new(&gRadioSpinalInstanceRaw) PlatformSpiRadioSpinel();
+}
+
+extern "C" void platformRadioInit(void)
+{
+	InitPlatformRadioSpinelInstance();
+	RadioSpinelInstance().GetSpinelInterface().Init();
+	RadioSpinelInstance().Init(IS_ENABLED(CONFIG_OPENTHREAD_HOSTPROCESSOR_RESET_RADIO), false, false);
+}
+
+extern "C" void platformRadioDeinit(void)
+{
+	RadioSpinelInstance().Deinit();
+}
+
+extern "C" void platformRadioProcess(otInstance *aInstance)
+{
+	RadioSpinelContext ctx = {
+		.mInstance = aInstance,
+	};
+
+	RadioSpinelInstance().Process(ctx);
+}
+
+static inline otExtAddress otExtAddressLittleEndian(const otExtAddress *aAddress)
+{
+	otExtAddress addr;
+
+	for (size_t i = 0; i < sizeof(addr); i++) {
+		addr.m8[i] = aAddress->m8[sizeof(addr) - 1 - i];
+	}
+
+	return addr;
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+
+	SuccessOrDie(RadioSpinelInstance().GetIeeeEui64(aIeeeEui64));
```

Comment:
Not sure if the `SuccessOrDie` is the best approach to "fail".

Reply:
Some caveat here, the included `radio_spinel_impl.hpp` also uses these functions a lot.